js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx

```diff
@@ -125,6 +136,92 @@ export const AdvancedSettings = ({ isOIDC, isSAML }: AdvancedSettingsProps) => {
       />
       <SwitchField field="config.hideOnLoginPage" label="hideOnLoginPage" />
 
+      {(!isSAML || isOIDC) && (
+        <FormGroupField label="filteredByClaim">
+          <Controller
+            name="config.filteredByClaim"
+            defaultValue="false"
+            control={control}
+            render={({ field }) => (
+              <Switch
+                id="filteredByClaim"
+                label={t("common:on")}
+                labelOff={t("common:off")}
+                isChecked={field.value === 'true'}
+                onChange={(value) => {
+                    const v=""+value
+                    field.onChange(v)
```

Comment:
Instead of declaring a new variable and concatenating a string to do a cast, you could call [`toString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean/toString) instead.
```suggestion
                    field.onChange(value.toString())
```